### ISSUE TITLE
[Docs] Do we need a "Debian Policy" style document for Charms and Libraries

### ISSUE THREAD

**Issue by pengale** [canonical/operator#766]
As we develop best practices around writing charms with the charmed operator framework, it would be nice to collect CharmHub listing requirements, along with recommended best practices into a single place.

The document might cover:
- Charm naming best practices
- Best practices around library naming, and where to store library source code.
- An explanation of the way that Charms are exposed (or not) on Charmhub, along with minimal requirements for a charm to be publicly listed.

Basically, any time that we have a discussion and agree on a best practice, it should make it into this doc.

**Comment by jnsgruk** [comment]
We do have this: https://juju.is/docs/sdk/naming

We also have the style guide https://juju.is/docs/sdk/styleguide 

But we might be able to do better around linking these and advertising them - and I do think better docs around the conditions of making a charm publicly visible/searchable would be very useful indeed.

**Comment by rwcarlsen** [comment]
I know there was some discussion about charmhub visibility at the last roadmap meeting.  But is there somewhere where decisions/direction around that has been recorded/distilled?

**Comment by pengale** [comment]
> I know there was some discussion about charmhub visibility at the last roadmap meeting. But is there somewhere where decisions/direction around that has been recorded/distilled?

IIRC, the decision was that search results would continue to include only listed charms, but that the charm teams within Canonical would be more pro-active about getting their charms listed. Listing is a matter of dropping into the discourse, and presenting a case for the charm being listed.

One of the TODOs from the meeting was the write up more complete docs about the process.

**Comment by mcjaeger** [comment]
Hello @pengale thanks for pointing out it is useful. it is already in progress. (and I thought you might have been aware of it?)

There are two points to this points:

1. Which are the relevant criteria?
2. Then, how do we apply these criteria?

It makes sense to start with the first part, and have had this discussion about qualities sand capabilities we see important when creating charmed operators, and we shared these with the community already. It is very obvious that we need to be transparent about it and encourage collaboration.

We call this publication checklist (= minimal points to care for as guidance to Charm developers) and we had two community workshops working on these the points there - to get also the important feedback. One workshop took even place today:

https://discourse.charmhub.io/t/community-workshop-publication-checklist-for-charms-part-deux/6516/3

the previous one on April 29th:

https://discourse.charmhub.io/t/community-workshop-publication-checklist-for-charms/6277/6

I wish we could have a third community workshop in July, but it makes sense to integrate the feedback from the past two meetings and then continue with a third meeting.

**Comment by benhoyt** [comment]
I think our docs and process on these have been publicized fairly widely now, so closing this. We can always open more specific issues if appropriate.
